src/box/txn_limbo.c

```diff
@@ -509,7 +541,9 @@ txn_limbo_read_confirm(struct txn_limbo *limbo, int64_t lsn)
 		 */
 		assert(e->txn->signature >= 0);
 		txn_limbo_complete(e->txn, true);
+		last_confirm_lsn = e->lsn;
 	}
+	txn_limbo_update_confirmed_entry_after_confirm(limbo, last_confirm_lsn);
```

Comment:
Below is a check:
```C
	if (limbo->confirmed_lsn < lsn) {
		limbo->confirmed_lsn = lsn;
		vclock_follow(&limbo->confirmed_vclock, limbo->owner_id, lsn);
	}
```
Why do you update the confirmed entry even when `confirmed_lsn` didn't change?

Reply:
`confirmed_lsn` might not have changed, but it could be that `lsn == confirmed_entry`. In this case, `confirmed_entry` needs to be updated.
https://github.com/Astronomax/tarantool/blob/6eee079a537a40c9721f6ca8a9fe616748f3b5c6/src/box/txn_limbo.c#L531-L534